library/src/main/java/com/novoda/downloadmanager/ByteBasedStorageRequirementRule.java

```diff
@@ -0,0 +1,32 @@
+package com.novoda.downloadmanager;
+
+import java.io.File;
+
+public final class ByteBasedStorageRequirementRule implements StorageRequirementRule {
+
+    private final StorageCapacityReader storageCapacityReader;
+    private final long bytesRemainingAfterDownload;
+
+    public static ByteBasedStorageRequirementRule withBytesOfStorageRemaining(long bytesRemainingAfterDownload) {
```

Comment:
I'd hide this public constructor and the class itself, then from the usage API point of view you can create a unique `StorageRequirementRuleFactory` that returns an implementation of `StorageRequirementRule`.

This has two advantatges:
- All the internal implementation of the download manager keeps being private to the library
- Only the interface is public to the client
- The user has an intuitive way to find out if we provide implementations for the required interface at `. withStorageRequirementRules`

Reply:
fc19d23